addons/mail/models/mail_template.py

```diff
@@ -89,6 +89,34 @@ def default_get(self, fields):
     can_write = fields.Boolean(compute='_compute_can_write',
                                help='The current user can edit the template.')
 
+    def _geneate_template_for_constrain(self):
+        self.ensure_one()
+        model = self.model_id.model
+        if not model:
+            return
+        record = self.env[model].search([], limit=1)
+        if not record.ids:
+            return
+        mail_template_fields = [
+            'body_html',
+            'subject',
+            'email_cc',
+            'email_from',
+            'email_to',
+            'partner_to'
+        ]
+        return self._generate_template(record.ids, mail_template_fields)
+
+    @api.constrains('body_html', 'subject', 'email_cc', 'email_from', 'email_to', 'partner_to')
+    def _constrain_template_rendering(self):
+        try:
+            for template in self:
+                template._geneate_template_for_constrain()
+        except AccessError:
+            raise
+        except:
+            raise ValidationError(_("There are issues in the Architecture of this template"))
```

Comment:
Could be nice to have something more precise. I believe the original error contains the reason for the failure at this point? Could be nice if we could forward that information